### PR TITLE
api extensions: loadable plugin framework description

### PR DESCRIPTION
**Pull request by anisyanka**
This patch adds a small description of the design of
the plugin framework and shows APIs can be used for work with it.

Signed-off-by: Aleksandr Anisimov <a.anisimov@omprussia.ru>

### CODE REVIEW

File: architecture/globalplatform_api.rst [hunk 1]
```diff
@@ -788,8 +788,88 @@ The following entries shall be added to **Table 6-11**:
...
+plugins:
+
+.. code-block:: c
+
+    /*
+     * tee_invoke_supp_plugin() - invoke a tee-supplicant's plugin
+     * @uuid:       uuid of the plugin
+     * @cmd:        command for the plugin
+     * @sub_cmd:    subcommand for the plugin
+     * @buf:        data [for/from] the plugin [in/out]
```
Comment (jforissier) [inline architecture/globalplatform_api.rst:842]: `[to/from]` sounds better than `[for/from]`

**Review by jforissier — APPROVED** [review]
Hi @anisyanka,

Thanks for the update. Minor comments below, anyways:
`Acked-by: Jerome Forissier <jerome@forissier.org>`

File: architecture/globalplatform_api.rst [hunk 2]
```diff
@@ -788,8 +788,88 @@ The following entries shall be added to **Table 6-11**:
...
+
+The plugin framework is based on the RPC - ``OPTEE_MSG_RPC_CMD_PLUGIN``.
+This is a unified interface between TEE and plugins.
+TEE can only access the plugins by its UUID.
+
+After implementing this structure, a plugin has to be compiled as a shared
+object. The objects have to be placed into the directory defined
+by ``CFG_TEE_PLUGIN_LOAD_PATH``. This path can be set in the ``config.mk`` file
+in the optee_client_ git. By default it is set to
+**/usr/lib/tee-supplicant/plugins/**.
```
[the remark this answers is not in the document]
Comment (anisyanka) [reply architecture/globalplatform_api.rst:824]: No, there isn't :(
I can do separate PR to introduce something like `CFG_TEE_SUPP_PLUGIN_SUPPORT` for the ability to disable this feature.

[the remark this answers is not in the document]
Comment (jforissier) [reply architecture/globalplatform_api.rst:824]: It's OK, let's wait until someone actually needs it.

**Comment by anisyanka** [comment]
All comments have been addressed, tag added. Thanks for the review! :)

**Comment by jbech-linaro** [comment]
Thanks, I believe we can merge this now? From what I can see there are no pending patches in other gits, i.e., all functionality needed seems to have been merged, correct?

**Comment by jforissier** [comment]
@jbech-linaro fine with me

**Review by etienne-lms — APPROVED** [review]

**Comment by anisyanka** [comment]
Yes, that's right, all functionality has been merged

**Comment by anisyanka** [comment]
Tag added

**Pull request merged**